packages/model-viewer/src/features/scene-graph/image.ts

```diff
@@ -36,6 +36,8 @@ export const $applyTexture = Symbol('applyTexture');
  * Image facade implementation for Three.js textures
  */
 export class Image extends ThreeDOMElement implements ImageInterface {
+  [$sourceObject]: GLTFImage;
```

Comment:
I hadn't realized that TS had this way to narrow types; that's convenient!

Reply:
I think there's a few classes than can be cleaned up by narrowing types in this manner

Reply:
I think there's a few classes that can be cleaned up by narrowing types in this manner

Reply:
I think there's a few classes that can be cleaned up by narrowing types in this manner. A good opportunity for anyone that wants to get their feet wet with open source

Reply:
I think there's a few classes that can be cleaned up by narrowing types in this manner. A good opportunity for anyone who wants to get their feet wet with open source